AsyncDisplayKit/Details/ASPINRemoteImageDownloader.m

```diff
@@ -133,7 +134,18 @@ - (nullable id)downloadImageWithURL:(NSURL *)URL
                    downloadProgress:(ASImageDownloaderProgress)downloadProgress
                          completion:(ASImageDownloaderCompletion)completion;
 {
-  return [[self sharedPINRemoteImageManager] downloadImageWithURL:URL options:PINRemoteImageManagerDownloadOptionsSkipDecode completion:^(PINRemoteImageManagerResult *result) {
+  return [[self sharedPINRemoteImageManager] downloadImageWithURL:URL options:PINRemoteImageManagerDownloadOptionsSkipDecode progressDownload:^(int64_t completedBytes, int64_t totalBytes) {
+    if (downloadProgress == nil) { return; }
+    
+    /// If we're targeting the main queue and we're on the main thread, call immediately.
```

Comment:
I'm not sure if dispatching to the main thread is necessary here? I think the reason it is happening below is to prevent image flashing when the load happens synchronously. However I guess it doesn't hurt either and makes the behavior consistent… Maybe ignore me :)